config/terraform_cloud_workspace.go

```diff
@@ -0,0 +1,139 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+
+	"github.com/hashicorp/consul-terraform-sync/logging"
+)
+
+// TerraformCloudWorkspaceConfig is an enterprise-only configuration that controls
+// workspace attributes that are specific to a task.
+type TerraformCloudWorkspaceConfig struct {
+	ExecutionMode *string `mapstructure:"execution_mode"`
+	AgentPoolID   *string `mapstructure:"agent_pool_id"`
+	AgentPoolName *string `mapstructure:"agent_pool_name"`
+}
+
+func DefaultTerraformCloudWorkspaceConfig() *TerraformCloudWorkspaceConfig {
```

Comment:
Wasn't sure if this should be called `DefaultTerraformCloudWorkspaceConfig()` or `EmptyTerraformCloudWorkspaceConfig()`. Open to suggestions!

Reply:
I think Default is okay, makes it easier to add default values in the future should we extend the config